File: app/src/main/java/com/akari/ppx/xp/hook/code/misc/InnerOpenHook.java
package com.akari.ppx.xp.hook.code.misc;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.view.View;

import com.akari.ppx.BuildConfig;
import com.akari.ppx.xp.hook.code.SuperbHook;

import java.util.ArrayList;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XC_MethodReplacement;

import static de.robv.android.xposed.XposedBridge.invokeOriginalMethod;
import static de.robv.android.xposed.XposedHelpers.callMethod;
import static de.robv.android.xposed.XposedHelpers.callStaticMethod;
import static de.robv.android.xposed.XposedHelpers.findClass;
import static de.robv.android.xposed.XposedHelpers.getObjectField;

public class InnerOpenHook extends SuperbHook {
	@Override
	protected void onHook(ClassLoader cl) {
		hookMethod("com.sup.android.m_mine.utils.f", "a", new XC_MethodHook() {
			@Override
			protected void afterHookedMethod(MethodHookParam param) {
				ArrayList list = (ArrayList) param.getResult();
				if(list.size() == 0){
					return;
				}
				Object item = list.get(1);
				callMethod(item, "setSchemaNeedLogin", Boolean.FALSE);
				callMethod(item, "setTabName", "皮皮虾助手");
				callMethod(item, "setTabSchema", "akari://open");
				callMethod(item, "setExtra", "{\"icon_list\":null,\"alert\":false}");
				callMethod(item, "setType", 4);
			}
		});
		hookMethod("com.sup.android.m_mine.view.subview.e$1", "doClick", View.class, new XC_MethodReplacement() {
			@Override
			protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
				Object thiz = getObjectField(param.thisObject, "b");
				String schema = (String) callMethod(callStaticMethod(findClass("com.sup.android.m_mine.view.subview.e", cl), "b", thiz), "getTabSchema");
				if ("akari://open".equals(schema)) {
					Context context = ((View) getObjectField(thiz, "i")).getContext();
					Intent intent = new Intent();
					intent.setComponent(new ComponentName(BuildConfig.APPLICATION_ID, "com.akari.ppx.ui.home.HomeActivity"));
					context.startActivity(intent);
					return null;
				}
				return invokeOriginalMethod(param.method, param.thisObject, param.args);
			}
		});
	}
}

File: app/src/main/java/com/akari/ppx/xp/hook/misc/InnerOpenHook.kt
@file:Suppress("unused")

package com.akari.ppx.xp.hook.misc

import android.view.View
import com.akari.ppx.data.Const.APP_NAME
import com.akari.ppx.data.Const.TAB_SCHEMA
import com.akari.ppx.ui.MainActivity
import com.akari.ppx.utils.*
import com.akari.ppx.xp.Init.cl
import com.akari.ppx.xp.Init.myTabList
import com.akari.ppx.xp.Init.myTabListClass
import com.akari.ppx.xp.Init.myTabView
import com.akari.ppx.xp.Init.myTabViewClass
import com.akari.ppx.xp.hook.BaseHook

class InnerOpenHook : BaseHook {
    override fun onHook() {
        myTabListClass!!.hookAfterMethod(myTabList()) { param ->
            (param.result as ArrayList<*>).checkIf({ isNotEmpty() }) {
                with(get(1)) {
                    callMethod("setSchemaNeedLogin", false)
                    callMethod("setTabName", APP_NAME)
                    callMethod("setTabSchema", TAB_SCHEMA)
                    callMethod("setExtra", "{\"icon_list\":null,\"alert\":false}")
                    callMethod("setType", 4)
                }
            }
        }
        "${myTabViewClass!!.name}\$1".replaceMethod(cl, "doClick", View::class.java) { param ->
            val tabView = param.thisObject.getObjectField("b")
            when (myTabViewClass!!.callStaticMethod(myTabView(), tabView)
                ?.callMethodAs<String>("getTabSchema")) {
                TAB_SCHEMA -> {
                    myTabViewClass!!.declaredFields.find { f -> f.type == View::class.java }?.name.let { f ->
                        tabView?.getObjectFieldAs<View>(f)?.context?.let { MainActivity(it) }
                    }
                }
                else -> param.invokeOriginalMethod()
            }
        }
    }
}
